### PR TITLE
Add a way to choose which repository should be searched 

### PR DESCRIPTION
**Pull request by luispereira**
So now it is possible to search only maven, jcenter or mavencentral repositories, using "-r" or "--repository". Additionally the user can search more than 1 repository by doing 
       `alfi "repository_name" -r m -r jc`
This will search in maven and jcenter repositores

### CODE REVIEW

**Comment by luispereira** [comment]
Is necessary a test for the search with repository? 
Also is necessary the exclude of SSL in order to run this on Windows without any SSL patch?

**Comment by cesarferreira** [comment]
Yeah, that's be great mate :)
It'd increase the code quality

File: lib/alfi/cli.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def exit_with(message)
   end
 
   def call(arguments)
+    @searchType = Array.new
```
Comment (ivan-leschinsky) [inline lib/alfi/cli.rb]: please, call variables according to ruby guides:

``` ruby
@search_type = []
```

File: lib/alfi/cli.rb [hunk 2]
```diff
@@ -20,7 +21,7 @@ def call(arguments)
 
     exit_with("Missing query parameter\n".red + @opt_parser.help) unless search_param
 
-    Alfi::Search.new.call(search_param)
+    Alfi::Search.new.call(search_param, @searchType)
```
Comment (ivan-leschinsky) [inline lib/alfi/cli.rb]: @search_type in the all project please 😄

File: spec/alfi_spec.rb [hunk 3]
```diff
@@ -2,14 +2,20 @@
...
 describe Alfi do
   it 'Test the query_url' do
-    str = Alfi::Providers::Maven.new('picasso').query_url('picasso')
+    str = Alfi::Providers::Maven.new('picasso', Array.new).query_url('picasso')
     expect(str).to eq 'http://search.maven.org/solrsearch/select?q=picasso&rows=350&wt=json'
   end
 
+  it 'Test search with repository' do
+    maven = Alfi::Search.new
+    maven.call('sPref', Array.new(1, "maven"))
```
Comment (ivan-leschinsky) [inline spec/alfi_spec.rb]:
``` ruby
maven.call('sPref', ['maven'])
```

File: spec/alfi_spec.rb [hunk 4]
```diff
@@ -2,14 +2,20 @@
...
+    expect(maven.total_results).to be >= 1
+  end
+
   context 'suggestions' do
     it 'should return 0 suggestions' do
       VCR.use_cassette('search_active_android_with_no_suggestions') do
-        maven_provider = Alfi::Providers::Maven.new('active-android')
+        maven_provider = Alfi::Providers::Maven.new('active-android', Array.new)
```
Comment (ivan-leschinsky) [inline spec/alfi_spec.rb]:
``` ruby
maven_provider = Alfi::Providers::Maven.new('active-android', [])
```

File: lib/alfi/search.rb [hunk 5]
```diff
@@ -29,4 +29,8 @@ def call(search_param)
     return if $suggestions.empty?
     puts "Did you mean: #{$suggestions.join(', ').yellow}"
   end
+
+  def total_results()
```
Comment (ivan-leschinsky) [inline lib/alfi/search.rb]: without braces please

``` ruby
def total_results_count
```

File: lib/alfi/search.rb [hunk 6]
```diff
@@ -9,14 +9,14 @@ def exit_with(message)
...
+  def call(search_param, searchType)
     return puts 'The search needs 3+ characters'.red if search_param.size < 3
     puts "Searching...\n"
 
-    Alfi::Providers.all.each { |cc| cc.new(search_param).call }
+    Alfi::Providers.all.each { |cc| cc.new(search_param, searchType).call }
 
     exit_with('No results'.red) if $result_list.empty? && $suggestions.empty?
-    num_results = $result_list.count { |r| r.strip[0] != '#' }
+    num_results = total_results()
```
Comment (ivan-leschinsky) [inline lib/alfi/search.rb]:
``` ruby
num_results = total_results_count
```

**Comment by luispereira** [comment]
Thanks @ivan-leschinsky, I still learning to proper write ruby, sorry about this.

**Comment by ivan-leschinsky** [comment]
@luispereira Don't worry about this, people always have something to learn 😃

**Comment by luispereira** [comment]
@cesarferreira What do you think now?

**Comment by cesarferreira** [comment]
## Regardig this part of the README:

Search for `something` by repository

``` bash
alfi picasso -r m
```

This will search picasso only on maven, you can also define multiple repositories like:

``` bash
alfi picasso -r mc -r jc
```

---

I'd prefer having a more explicit README, like: 

``` bash
alfi picasso --repository maven
alfi picasso --repository jcenter
alfi picasso --repository mavenCentral --repository maven
```

Regarding the syntax, I personally don't like the abbreviations for the repositories
I don't find it very clear to the users, everyone knows `maven` and `jcenter`, but M or JC it's not that clear at all ;)

File: lib/alfi/cli.rb [hunk 7]
```diff
@@ -10,6 +10,7 @@ def exit_with(message)
   end
 
   def call(arguments)
+    @searchType = []
```
Comment (ivan-leschinsky) [inline lib/alfi/cli.rb]: Please change `@searchType` to the `@search_type` in this file

File: lib/alfi/search.rb [hunk 8]
```diff
@@ -9,14 +9,14 @@ def exit_with(message)
...
+  def call(search_param, search_type)
     return puts 'The search needs 3+ characters'.red if search_param.size < 3
     puts "Searching...\n"
 
-    Alfi::Providers.all.each { |cc| cc.new(search_param).call }
+    Alfi::Providers.all.each { |cc| cc.new(search_param, search_type).call }
 
     exit_with('No results'.red) if $result_list.empty? && $suggestions.empty?
-    num_results = $result_list.count { |r| r.strip[0] != '#' }
+    num_results = total_results_count()
```
Comment (ivan-leschinsky) [inline lib/alfi/search.rb]: Please don't use brackets here. Just `num_results = total_results_count`

File: lib/alfi/search.rb [hunk 9]
```diff
@@ -29,4 +29,8 @@ def call(search_param)
     return if $suggestions.empty?
     puts "Did you mean: #{$suggestions.join(', ').yellow}"
   end
+
+  def total_results_count()
```
Comment (ivan-leschinsky) [inline lib/alfi/search.rb]: Please don't use brackets here. Just `def total_results_count`

**Comment by luispereira** [comment]
Thanks guys, hope it is all good now ;)

**Comment by cesarferreira** [comment]
just pushed `0.4.5` to rubygems :) thanks to all, specially to @luispereira

**Pull request merged**